### DISCUSSION TITLE
Force Loss Convergence

### DISCUSSION THREAD

**Discussion by caolz** [deepmodeling/deepmd-kit#709] [Q&A]
I am training a model using the "se_a" type.
{
    "_comment": " model parameters",
    "model": {
        "type_map":     ["Au"],
	"descriptor" :{
	    "type":		"se_a",
	    "sel":		[25],
	    "rcut_smth":	5.80,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	12,
	    "seed":		3246556277,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1978363284,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	5000,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"start_pref_e": 0.05,
	"limit_pref_e":	1,
	"start_pref_f":	1000,
	"limit_pref_f":	1,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "_comment": " traing controls",
    "training" : {
	"systems":	["../1-data/Au16", "../1-data/Au17", "../1-data/Au18", "../1-data/Au19", "../1-data/Au20", "../1-data/Au21", "../1-data/Au22", "../1-data/Au23", "../1-data/Au24", "../1-data/Au25"],
	"set_prefix":	"set",    
	"stop_batch":	1000000,
	"batch_size":	"auto",

	"seed":		2662107775,

	"_comment": " display and restart",
	"_comment": " frequencies counted in batch",
	"disp_file":	"lcurve.out",
	"disp_freq":	100,
	"numb_test":	10,
	"save_freq":	1000,
	"save_ckpt":	"model.ckpt",
	"load_ckpt":	"model.ckpt",
	"disp_training":true,
	"time_training":true,
	"profiling":	false,
	"profiling_file":"timeline.json",
	"_comment":	"that's all"
    },

    "_comment":		"that's all"
}
I only modify sel, it has been changed 6 times, which are 25,100,233,500,800,1000.
I drew the Loss of energy and force respectively.

sel = 25
![25-eng-loss](https://user-images.githubusercontent.com/49866647/120964303-b3786b00-c795-11eb-956a-2072951d9895.jpg)
![25-for-loss](https://user-images.githubusercontent.com/49866647/120964327-c1c68700-c795-11eb-9e5b-e9d3567597f7.jpg)

sel = 100
![100-eng-loss](https://user-images.githubusercontent.com/49866647/120964416-dd319200-c795-11eb-8354-5dd825c90de8.jpg)
![100-for-loss](https://user-images.githubusercontent.com/49866647/120964422-e15daf80-c795-11eb-9577-93f0cd5cc099.jpg)

sel = 233
![233-eng-loss](https://user-images.githubusercontent.com/49866647/120964440-e9b5ea80-c795-11eb-8ce8-254868356ead.jpg)
![233-for-loss](https://user-images.githubusercontent.com/49866647/120964446-eae71780-c795-11eb-9e6c-ed9bce9dc82c.jpg)

sel = 500
![500-eng-loss](https://user-images.githubusercontent.com/49866647/120964521-08b47c80-c796-11eb-85a1-bf69d5dffedb.jpg)
![500-for-loss](https://user-images.githubusercontent.com/49866647/120964642-31d50d00-c796-11eb-846e-42d8982dccd8.jpg)

sel = 800
![800-eng-loss](https://user-images.githubusercontent.com/49866647/120964654-38fc1b00-c796-11eb-918f-7f36be8c5f4e.jpg)
![800-for-loss](https://user-images.githubusercontent.com/49866647/120964663-3b5e7500-c796-11eb-80bb-cc6e47c9d029.jpg)

sel = 1000
![1000-eng-loss](https://user-images.githubusercontent.com/49866647/120964680-41ecec80-c796-11eb-8207-2874ab762f82.jpg)
![1000-for-loss](https://user-images.githubusercontent.com/49866647/120964684-444f4680-c796-11eb-9628-91c59624533f.jpg)

How can I solve it?

**Comment by njzjz** [comment]
You may check your data - I guess it's very likely to be incorrect. The energy and force seem to be inconsistent.

**Reply by caolz** [reply]
Can you help me check if my unit conversion is wrong? I extracted data from DMol3. The energy unit is 1 Ha = 27.2114 eV, the coordinate unit is 1 a.u. = 0.529177249 Angstrom, and the force unit is 1 Ha/Bohr = 51.42208619083232 eV/Angstrom. Do I need to use scientific notation after conversion?

**Reply by njzjz** [reply]
Did you use **negative** energy derivative？ Force is a negative gradient of the potential energy.

**Reply by caolz** [reply]
![image](https://user-images.githubusercontent.com/49866647/121327447-0729b680-c946-11eb-9a41-598b627e22f5.png)
As you can see, starting from the third row and third column, I extract the coordinates and force, and convert the units.

**Reply by njzjz** [reply]
The third column is energy derivatives but not forces. Force is the **negative** energy derivative.

**Reply by caolz** [reply]
Thank you for your help!
